storage/remote/otlptranslator/prometheusremotewrite/helper.go

```diff
@@ -518,6 +527,14 @@ func createLabels(name string, baseLabels []prompb.Label, extras ...string) []pr
 	return labels
 }
 
+// addTypeAndUnitLabels appends type and unit labels to the given labels slice if the setting is enabled.
```

Comment:
"if the setting is enabled" isn't the case, because this function doesn't check the setting:
```suggestion
// addTypeAndUnitLabels appends type and unit labels to the given labels slice.
```